### PR TITLE
Cache charmcraft pack container, skip unstable tests except on schedule

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**
Ported from https://github.com/canonical/mysql-k8s-operator/pull/146

### CODE REVIEW

File: tests/unit/test_charm.py [hunk 1]
```diff
@@ -344,7 +345,7 @@ def test_mongodb_relation_joined_all_replicas_not_ready(self, connection):
     @patch_network_get(private_address="1.1.1.1")
     @patch("ops.framework.EventBase.defer")
     @patch("charm.MongoDBConnection")
-    @patch("lib.charms.mongodb.v0.mongodb.MongoClient")
+    @patch("charms.mongodb.v0.mongodb.MongoClient")
```
Comment (MiaAltieri) [inline tests/unit/test_charm.py:348]: Sorry for the noob question, but why this change?

**Review by MiaAltieri — COMMENTED** [review]
Carl this looks awesome! Thank you so much for your valuable contribution here! There are some changes I am confused about that I am not sure how they are related to the `cache`ing of `charmcraft pack`. Is it possible you have a different formatter and re-formatted some files accidentally? 

This is something I am very new to, so if they are related you can totally ignore this comment. But I would love a small explanation so I can expand my understanding :)

**Review by MiaAltieri — APPROVED** [review]
thanks for the explaination. LGTM

**Review by delgod — APPROVED** [review]
